include/lib/queue.h

```diff
@@ -4,9 +4,17 @@
 
 #include "stdlib/stdtypes.h"
 
+/** Data structure for a node */
+struct node {
+    void *data;
+    struct node *next;
+};
+
 /** Data structure for a generic FIFO queue */
 struct queue {
-    // TODO
+    struct node *front;
```

Comment:
Fields in structs should be documented with their uses to keep up maintainability.